src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Parser/SPDX30Parser.cs

```diff
@@ -114,32 +116,36 @@ public SPDX30Parser(
     /// <exception cref="ParserException"></exception>
     public ParserStateResult? Next()
     {
-        ParserStateResult? result;
+        ParserStateResult? result = null;
         do
         {
-            result = this.parser.Next();
-            if (result is not null)
+            result = parser.Next();
+
+            if (result is not null && result.Result is not null)
```

Comment:
Consider null propagation?
```suggestion
            if (result?.Result is not null)
```